src/Domain/Block.cpp

```diff
@@ -19,15 +21,29 @@ Block<VolumeDim>::Block(
     std::unique_ptr<domain::CoordinateMapBase<Frame::Logical, Frame::Inertial,
                                               VolumeDim>>&& stationary_map,
     const size_t id,
-    DirectionMap<VolumeDim, BlockNeighbor<VolumeDim>> neighbors) noexcept
+    DirectionMap<VolumeDim, BlockNeighbor<VolumeDim>> neighbors,
+    DirectionMap<VolumeDim, std::unique_ptr<
+                                domain::BoundaryConditions::BoundaryCondition>>
+        external_boundary_conditions) noexcept
     : stationary_map_(std::move(stationary_map)),
       id_(id),
       neighbors_(std::move(neighbors)) {
   // Loop over Directions to search which Directions were not set to neighbors_,
   // set these Directions to external_boundaries_.
   for (const auto& direction : Direction<VolumeDim>::all_directions()) {
     if (neighbors_.find(direction) == neighbors_.end()) {
-      external_boundaries_.emplace(std::move(direction));
+      if (not external_boundary_conditions.contains(direction)) {
+        // Fill with a nullptr for now so we don't break existing code.
+        external_boundary_conditions_[direction] = nullptr;
+        // We cannot enable this error message until boundary conditions are
+        // enabled everywhere in the code, unfortunately.
+        // ERROR("Couldn't find boundary condition for external boundary "
+        //       << direction);
```

Comment:
I think you can already test the reverse case though: Make sure the user didn't pass a boundary condition for a direction that's actually internal